smart_campus/app/views.py

```diff
@@ -618,3 +639,390 @@ def reward_add_page(request):
+        return HttpResponseForbidden()
+
+    station.delete()
+
+    if request.user.can(Permission.ADMIN):
+        stations = Station.objects.all()
+    else:
+        stations = Station.objects.filter(owner_group=request.user.group)
+
+    station_data = [
+        {
+            'id': station.id,
+            'name': station.name,
+            'primary_image': StationImage.objects.filter(
+                    station=station,
+                    is_primary=True
+                ).first(),
+            'category': station.category,
+            'beacon': Beacon.objects.filter(
+                station=station
+            ).first()
+        }
+        for station in stations
+    ]
+
+    context = {
+        'email': request.user.email,
+        'stations': station_data,
+        'categories': StationCategory.objects.all()
+    }
+
+    return render(request, 'app/station_list.html', context)
+
+
+@login_required
+def travelplan_list_page(request):
+    # list the travelplan on teh page
+    context = {
+        'categories': StationCategory.objects.all(),
+        'email': request.user.email,
+        'travelplans': TravelPlan.objects.all().order_by('id')
+    }
+
+    return render(request, 'app/travelplan_list_page.html', context)
+
+
+@login_required
+def travelplan_add_page(request):
+    # add the travelplan using partialtravelplanform
+    if request.method == 'POST':
+        travelplan_form = PartialTravelPlanForm(request.POST, request.FILES)
+
+        if travelplan_form.is_valid():
+            travelplan_form.save()
+
+            context = {
+                'categories': StationCategory.objects.all(),
+                'email': request.user.email,
+                'travelplans': TravelPlan.objects.all()
+            }
+
+            return HttpResponseRedirect('/travelplans/')
+    else:
+        travelplan_form = PartialTravelPlanForm()
+
+    context = {
+        'categories': StationCategory.objects.all(),
+        'email': request.user.email,
+        'form': travelplan_form
+    }
+
+    return render(request, 'app/travelplan_add_page.html', context)
+
+
+@login_required
+def travelplan_edit_page(request, pk):
+    # edit the travelplan content using travelplanform
+
+    # use pk to get the travelplan
+    travelplan = get_object_or_404(TravelPlan, pk=pk)
+
+    if request.method == 'POST':
+        # instance varible to specify the travelplan being edited
+        travelplan_form = PartialTravelPlanForm(
+                                request.POST,
+                                request.FILES,
+                                instance=travelplan
+                            )
+        print(request.POST)
+        if travelplan_form.is_valid():
+            edited_travelplan = travelplan_form.save()
+
+            json_order = json.loads(request.POST['order'])
+
+            if not json_order:
+                for travelplan_station in TravelPlanStations.objects.filter(
```

Comment:
**DO NOT** break lines like this.

Even `for travelplan_station in TravelPlanStations.objects.filter(travelplan_id=pk):` is far more better than this.

Reply:
But the for loop can't be divided, right?
What is the even better way to indent here?

Reply:
@Lee-W